pkg/archive/pom.go

```diff
@@ -0,0 +1,78 @@
+package archive
+
+import (
+	"encoding/xml"
+	"regexp"
+)
+
+type Build struct {
+	Plugins struct {
+		Plugin []struct {
+			ArtifactID    string `xml:"artifactId"`
+			GroupID       string `xml:"groupId"`
+			Configuration struct {
+				// This is the property that contains the name of the git properties file.
+				GenerateGitPropertiesFilename string `xml:"generateGitPropertiesFilename"`
+			} `xml:"configuration"`
+		} `xml:"plugin"`
+	} `xml:"plugins"`
+}
+
+type Project struct {
+	Build Build `xml:"build"`
+
+	Profiles struct {
+		Profile []struct {
+			Build Build `xml:"build"`
+		} `xml:"profile"`
+	} `xml:"profiles"`
+}
+
+func getAllGitPropertiesFileInPom(pom []byte) []string {
+	// This function returns all the git properties files that are generated by git-commit-id-maven-plugin
+	// in the given pom file.
+	// In theory, there can be multiple git properties files in a single pom file so we return a list of them.
+	mavenProject := Project{}
```

Comment:
nit: I think parsing should probably be handled elsewhere and this function accept the already-parsed project. that way the caller can decide how to handle this error

Reply:
Handled in `archive/manifest/pom.go`.